### ISSUE TITLE
[ASK] Sequential Recommender (SLi_Rec) Questions

### ISSUE THREAD

**Issue by elogicsal** [microsoft/recommenders#1040]
### Description
When we create a dataset for the predict function in SLi_Rec do we need to start with a label i.e. 1 or 0 and what timestamp should we use? If we use a future timestamp does the model provide a probability based on the timestamp? How about if we use the timestamp for NOW() or a timestamp in the past?

What are the tradeoffs/benefits for min_sequence_len and max_sequence_len in terms of network complexity and compute memory and time?

Is there a way to influence the alpha parameter that balances short-term and long-term?

What were the RAM requirements for the full Amazon dataset and the training data file size?

Any tips on providing the model with a more complex representation for Category. I see that under the hood there is an embedding happening, wondering if there is a way to include a combination of columns.

**Comment by yueguoguo** [comment]
@Leavingseason FYI

**Comment by Leavingseason** [comment]
Hi there sorry that it seems I cannot receive notifications from github issues... in the future if you encounter something need my attention, welcome to send emails to me via jianxun.lian@outlook.com   @yueguoguo 
To @ghost, 
(1) for the predict function in SLi_Rec, yes there is a label placeholder, but it will not be used so you can make up any one for each line; the timestamp is suggested to be time you make inference, such as NOW() 
(2) max_sequence_len  is a balance for computational complexity and accuracy, you can determine a suitable according to your dataset. min_sequence_len  is a filter that we remove noisy instances.
(3) for our model we hope it is learned from the training data; but as stated in our paper experiment section, yes, we can replace it with some fixed value
(4) I think it depends on what types of attributes the item contains in your dataset. If the dataset has only itemID and categoryID just like the Amazon dataset, I prefer to use a simple concatenation of embedding lookup, while the embeddings can be pretrained from an item2item task or from a item-to-category topology. If items contain diverse attributes, you can try some fusion models like DNN and Transformer.

**Comment by rakshitsakhuja** [comment]
@Leavingseason

I was trying to reproduce the results from SLI-Rec, and after going through all the gihtub issues regrading SLI-rec. I still had few questions.

1. What are the minimal steps i need to do to return results for an item, like if i give user a and item b , how can i generate sequence of items
2.  I understood from #1254 that  we need to enumerate all the items you need to make prediction for the user as <user, item> pair
3. For this to work do i need to write another evaluate function or existing run_eval function would do the work, but then it would need data in the test file type of format only.
4. Also test file has negatives and positives and then metrices are calculated, so in my case as i want to generate sequence of next items, shall i give place holder for label as 0 for all?

**Comment by AndreasKarasenko** [comment]
@rakshitsakhuja did you make any progress with this?
I also would like to know how to get the next k best items per user, but I'm a bit confused on how to do this ...
